src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -1920,6 +1937,18 @@ internal void Close()
             FullCleanup();
             LogMessages.ClosedPhysicalConnection(ConnectionLogger, Host, Port, Database, UserFacingConnectionString, Id);
         }
+        finally
+        {
+            try
+            {
+                _semaphore.Release();
+                _semaphore.Dispose();
```

Comment:
It seems like if we're not closing the connector (but, for example, breaking it while opening) the semaphore is not going to get disposed.

Reply:
We must guarantee that the `Dispose`/`Close` method is correctly called once a connector is not usable/used anymore.
Should we check everywhere that this statement is correctly enforced?